format.go

```diff
@@ -57,7 +57,8 @@ func (d *Decimal) Append(buf []byte, fmt byte) []byte {
 		return append(buf, "unknown"...)
 	}
 
-	digits := d.Coeff.String()
+	var scratch [16]byte
```

Comment:
Does `scratch` escape to the heap? If it does, could we Append directly to `buf`? Or temporarily use the spare capacity at the end of `buf`? `fmtE` and `fmtF` look like they might make that tricky.

Reply:
Doesn't look like it;
I couldn't find scratch itself in the output, but 
```
./format.go:107:35: digits does not escape
```
refers to digits argument to fmtE/f functions -- so, no I don't think scratch escapes.